gensim/scripts/segment_wiki.py

```diff
@@ -263,25 +271,22 @@ def get_texts_with_sections(self):
         total_articles, total_sections = 0, 0
         page_xmls = extract_page_xmls(self.fileobj)
         pool = multiprocessing.Pool(self.processes)
-        # process the corpus in smaller chunks of docs, because multiprocessing.Pool
-        # is dumb and would load the entire input into RAM at once...
-        for group in utils.chunkize(page_xmls, chunksize=10 * self.processes, maxsize=1):
-            for article_title, sections in pool.imap(segment, group):  # chunksize=10):
-                # article redirects are pruned here
-                if any(article_title.startswith(ignore + ':') for ignore in IGNORED_NAMESPACES):  # filter non-articles
-                    skipped_namespace += 1
-                    continue
-                if not sections or sections[0][1].lstrip().lower().startswith("#redirect"):  # filter redirect
-                    skipped_redirect += 1
-                    continue
-                if sum(len(body.strip()) for (_, body) in sections) < self.min_article_character:
-                    # filter stubs (incomplete, very short articles)
-                    skipped_length += 1
-                    continue
-
-                total_articles += 1
-                total_sections += len(sections)
-                yield (article_title, sections)
+        for article_title, sections in pool.imap_unordered(segment, page_xmls, chunksize=10):
```

Comment:
This is not a good idea -- if the producer/consumer speed is mismatched, the pool will fill up internal queues and run out of memory (too many tasks in the finished queue / too many tasks in the input queue).

Reply:
Actually, the main process is still the busiest as there is a parsing xml and collecting results. Children processes are weakly loaded.

I don't think that article order is really important (but can back to `imap` in other case).

Reply:
@piskvorky as I remember, the order of articles have no sense (no semantic), the only order of sections (in one article) have sense. Let me check.

Reply:
Yes, that's a side note, I also don't think it's important. The queues filling up without control (unbounded size of input/output queues) is the primary concern here. I don't remember exactly the code exactly, but I have no doubt the chunking was added for a reason.

Reply:
@piskvorky probably `unordered` gives the performance boost, but need to check it.